clang-tools-extra/clang-tidy/readability/UseStdMinMaxCheck.cpp

```diff
@@ -0,0 +1,139 @@
+//===--- UseStdMinMaxCheck.cpp - clang-tidy -------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "UseStdMinMaxCheck.h"
+#include "../utils/ASTUtils.h"
+#include "clang/AST/ASTContext.h"
+#include "clang/ASTMatchers/ASTMatchFinder.h"
+#include "clang/Lex/Preprocessor.h"
+#include <optional>
+
+using namespace clang::ast_matchers;
+
+namespace clang::tidy::readability {
+
+UseStdMinMaxCheck::UseStdMinMaxCheck(StringRef Name, ClangTidyContext *Context)
+    : ClangTidyCheck(Name, Context),
+      IncludeInserter(Options.getLocalOrGlobal("IncludeStyle",
+                                               utils::IncludeSorter::IS_LLVM),
```

Comment:
```
void UseStdMinMaxCheck::storeOptions(ClangTidyOptions::OptionMap &Opts) {
  Options.store(Opts, "IncludeStyle", IncludeInserter.getStyle());
  Options.store(Opts, "AlgorithmHeader",
                Options.get("AlgorithmHeader", "<algorithm>"));
}
```
Here, the AlgorithmHeader field is initialized in the member initializer list of the constructor using Options.get("AlgorithmHeader", "<algorithm>")
is this not ok??

Reply:
The variable `AlgorithmHeader` is not even present in the snippet that you provided. So I highly doubt that it's initialized

Reply:
I am sorry , I think I misunderstood you, can you please suggest me where do i need to initialise it and how, sorry for such silly doubt.

Reply:
No worries, in the constructor of the check you should initialize the variable 
```
AlgorithmHeader(Options.get(...));
```
and then store the value of the variable in the `Store` method
```
Options.store(Opts, "AlgorithmHeader", AlgorithmHeader);
```

The logic is the same has `IncludeInserter`, the only difference is that you should use `Options.get` rather than `Options.getLocalOrGlobal`.

(To be transparent, I still think this option shouldn't exist since there is no reason to overwrite the header)